sound/soc/sof/sof-audio.c

```diff
@@ -880,6 +880,19 @@ struct snd_sof_widget *snd_sof_find_swidget(struct snd_soc_component *scomp,
 	return NULL;
 }
 
+struct snd_sof_widget *snd_sof_find_swidget_by_id(struct snd_soc_component *scomp, int id)
```

Comment:
It might be better to pass the `sdev` directly?
`struct snd_sof_widget *snd_sof_find_swidget_by_id(struct snd_sof_dev *sdev, int id)`